tests/test_colo_model_lsf.py

```diff
@@ -0,0 +1,232 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2023, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import sys
+
+import pytest
+
+from smartsim import Experiment
+from smartsim.entity import Model
+import smartsim.settings.base
+from smartsim.settings.lsfSettings import JsrunSettings
+
+
+is_mac = sys.platform == 'darwin'
+
+
+class ExpectationMet(Exception):
+    """Use this to break a test when we verify a call path is achieved"""
+    ...
+
+
+def show_expectation_met(*args, **kwargs):
+    raise ExpectationMet("mock._prep_colocated_db")
+
+
+def test_jsrun_prep(fileutils, coloutils, monkeypatch):
+    """Ensure that JsrunSettings prep method is executed as expected"""
+    monkeypatch.setattr(smartsim.settings.base, "expand_exe_path", lambda x: "/bin/{x}")
+    # mock the prep method to raise an exception that short circuits test when goal is met
+    monkeypatch.setattr(JsrunSettings, "_prep_colocated_db", show_expectation_met)
+
+    db_args = {"custom_pinning":[1]}
+    db_type = 'uds' # Test is insensitive to choice of db
+
+    exp = Experiment("colocated_model_lsf", launcher="lsf")
+
+    with pytest.raises(ExpectationMet, match="mock._prep_colocated_db") as ex:
+        run_settings = JsrunSettings("foo")
+
+        coloutils.setup_test_colo(
+            fileutils,
+            db_type,
+            exp,
+            db_args,
+            colo_settings = run_settings,
+        )
+
+    assert ex
```

Comment:
Will this `assert` ever be false? Surely the `pytest.raises` will throw if no exception is raised

Reply:
100. removing!